cmake/install/engine.json.in

```diff
@@ -1,8 +1,10 @@
 {
     "engine_name": "@LY_VERSION_ENGINE_NAME@",
     "restricted_name": "o3de",
-    "FileVersion": 1,
-    "O3DEVersion": "@LY_VERSION_STRING@",
+    "display_version": "@LY_DISPLAY_VERSION_STRING@",
+    "version": "@LY_VERSION_STRING@",
+    "api_versions": @LY_INSTALL_API_VERSIONS@,
+    "FileVersion": 2,
```

Comment:
Do we have json schema available for engine.json? 

Also is there a reason for case and spacing inconsistencies here (looks bad, but assume hard to fix)? O3DECopyrightYear vs api_versions?

I would also argue that FIleVersion should move to a metadata field ie $version or ideally $id and $schema ;)

Am surprised we do not have standard naming convention for json data we vend.

Reply:
We it should be all lower_underscore case, but it looks like SentenceCase was introduced for some fields at a point.

Reply:
I don't think we have a schema, but that's a good idea. I'll look at the other fields and see how they're referenced to see if we can change em to lower_underscore case without breaking anything.

Reply:
I did a scan for uses of the other fields and it looks fairly minor - will add sig-build team to the review though in case Jenkins uses any of these.  I've updated the field names to be consistent